docs/Administrator-Guide/Setting-Up-Clients.md

```diff
@@ -50,9 +50,10 @@ To install Gluster Native Client on RPM distribution-based systems
 2.  Ensure that TCP and UDP ports 24007 and 24008 are open on all
     Gluster servers. Apart from these ports, you need to open one port
     for each brick starting from port 49152 (instead of 24009 onwards as
-    with previous releases). The brick ports assignment scheme is now
-    compliant with IANA guidelines. For example: if you have
-    five bricks, you need to have ports 49152 to 49156 open.
+    with previous releases). The brick ports are now randomized, a port is
+    randomly selected within the range of base_port to max_port as defined
+    in glusterd.vol file and then assigned to the brick. For example: if you have
+    five bricks, you need to have at least 5 ports open within the given range.
```

Comment:
> For example: if you have five bricks, you need to have at least 5 ports open within the given range.

No. You need to open the entire range because the port selection is random and will change on each execution.

Reply:
What could be added to the documentation is that changing max_port could help reduce the range to avoid opening a big range of ports. We can also document how this is changed if not explained anywhere else.

Reply:
> No. You need to open the entire range because the port selection is random and will change on each execution.

Yes, but it will consume only 5 of them. So, even if we have the range as 1 - 100 (suppose) defined in the glusterd.vol and only 5 are open then in random probing it will find that it is occupied and it will not use that and look for another.

Though, limiting the open ports will increase execution time. And in that case, reducing the max_port range will be helpful.

Reply:
> Yes, but it will consume only 5 of them. So, even if we have the range as 1 - 100 (suppose) defined in the glusterd.vol and only 5 are open then in random probing it will find that it is occupied and it will not use that and look for another.

No, that won't happen. If glusterd chooses a port that happens to be blocked by the firewall, the machine will be able to use that port (so no EADDRINUSE error), but it won't be able to establish a connection. So clients won't be able to connect to that brick and the brick won't retry on another port.

Firewall should have the entire range open, but this means ~20000 ports. To make it more controlled, it would be better to reduce the range in the configuration.

Reply:
Oh okay. Thanks for the explanation, I didn't know that part.